canopen/objectdictionary/__init__.py

```diff
@@ -29,29 +29,33 @@ def export_od(od, dest: Union[str, TextIO, None] = None, doc_type: Optional[str]
     :rtype: str or None
     """
 
-    doctypes = {"eds", "dcf"}
-    if isinstance(dest, str):
-        if doc_type is None:
-            for t in doctypes:
-                if dest.endswith(f".{t}"):
-                    doc_type = t
-                    break
-
-        if doc_type is None:
-            doc_type = "eds"
-        dest = open(dest, 'w')
-    assert doc_type in doctypes
-
-    if doc_type == "eds":
-        from canopen.objectdictionary import eds
-        return eds.export_eds(od, dest)
-    elif doc_type == "dcf":
-        from canopen.objectdictionary import eds
-        return eds.export_dcf(od, dest)
-
-    # If dest is opened in this fn, it should be closed
-    if type(dest) is str:
-        dest.close()
+    opened_here = False
+    try:
+        doctypes = {"eds", "dcf"}
+        if isinstance(dest, str):
+            if doc_type is None:
+                for t in doctypes:
+                    if dest.endswith(f".{t}"):
+                        doc_type = t
+                        break
+
+            if doc_type is None:
+                doc_type = "eds"
```

Comment:
This could be a nice case of `for ... else`, if you want to save a few lines.

```suggestion
                for t in doctypes:
                    if dest.endswith(f".{t}"):
                        doc_type = t
                        break
                else:
                    doc_type = "eds"
```

Reply:
Ah nice. I've never quite gotten for..else into my fingers, but I like this 👌